src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -121,6 +121,53 @@ const DiffExprCompute = (props) => {
     );
   };
 
+  const renderError = () => {
+    if (!isFormValid) return <></>;
+    if (isVolcanoPlot && canRunDiffExpr() !== canRunDiffExprResults.TRUE) {
```

Comment:
This if statement is semantically weird. If `canRunDiffExpr` is true, then one would expect from the language that we can run differential expession. Why is the check `canRunDiffExprResults.TRUE` outside of that function? And why does it matter if differential expression is requested by a volcano plot or not? Why If it is a volcano plot, we would sometimes want to display an error? Shouldn't we be displaying that error if `canRunDiffExpr() !== canRunDiffExprResults.TRUE` is true, regardless of whether it is a volcano plot or not?

Reply:
ok I see. Then I would center the logic around whether we need to display the error not around whether it is a volcano plot, but whether pvalues are needed or not. I would pass in a variable for `needPvalues` or something like that instead of `isVolcanoPlot` and in the situation where we don't have p values and we need p values, I would return the error. To me, that sounds much more intuitive and it would support easier adding calls of diff expr from other places, in the case we need it. What do you think?

Reply:
Yep I'll rename the variable.